Update CHANGELOG

abhinav opened the pull request.

peter-edge commented:
Note that the `yarpc-protobuf-raw-response` header key changed to `rpc-protobuf-raw-response`, I deleted this per @breerly's suggestion

abhinav commented:
> Note that the yarpc-protobuf-raw-response header key changed to rpc-protobuf-raw-response, I deleted this per @breerly's suggestion

Changed comment to just that the wire repr. of proto changed and that this will break everyone.

kriskowal commented on CHANGELOG.md:
```diff
@@ -4,28 +4,35 @@ Releases
--   Pool inbound thrift request buffers to reduce allocations.
--   x/grpc: Remove NewInbound and NewSingleOutbound in favor of
-    functions on Transport.
+-   x/grpc: Remove `NewInbound` and `NewSingleOutbound` in favor of functions
```
functions/methods.

kriskowal commented on CHANGELOG.md:
```diff
@@ -4,28 +4,35 @@ Releases
 -------------------
 
+-   Different encodings can now register handlers under the same procedure
+    name.
```
Nice catch.
